File: zhixing_ai_coach/cloudfunctions/app/routes/payments.js
const { db } = require('../database')
const { generateOrderNo, getPayParams } = require('../services/wechatPay')
const config = require('../config')

exports.handle = async (event, userId) => {
  const method = event.httpMethod

  if (method === 'POST' && event.path.endsWith('/create-order')) {
    const memberId = event.queryStringParameters?.member_id
    if (!memberId) throw new Error('缺少member_id')

    const { data: members } = await db.collection('members').where({ _id: memberId, userId }).get()
    if (!members.length) throw new Error('会员不存在')

    const orderNo = generateOrderNo()
    const result = await db.collection('orders').add({
      data: {
        userId, memberId, orderNo, totalFee: config.PLAN_PRICE,
        status: 'pending', prepayId: `mock_${orderNo}`, transactionId: '',
        createdAt: db.serverDate(), updatedAt: db.serverDate(),
      },
    })

    await db.collection('orders').doc(result.id).update({
      data: { status: 'paid', paidAt: db.serverDate() },
    })

    const payParams = getPayParams(`mock_${orderNo}`)
    return { order_id: result.id, order_no: orderNo, total_fee: config.PLAN_PRICE, prepay_id: `mock_${orderNo}`, pay_params: payParams, simulated: true }
  }

  if (method === 'GET' && event.path.endsWith('/orders')) {
    const { data } = await db.collection('orders').where({ userId }).orderBy('createdAt', 'desc').get()
    return data.map(o => ({ id: o._id, order_no: o.orderNo, total_fee: o.totalFee, status: o.status, paid_at: o.paidAt || null, created_at: o.createdAt }))
  }

  throw new Error('Not found')
}

File: zhixing_ai_coach/backend/app/api/payments.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.ext.asyncio import AsyncSession
from sqlalchemy import select

from app.database import get_db
from app.models.user import User
from app.models.member import Member
from app.models.order import Order
from app.schemas.order import CreateOrderOut, OrderOut, PayNotify
from app.services.wechat_pay import generate_order_no, get_miniapp_pay_params
from app.config import settings
from app.utils.auth import get_current_user

router = APIRouter(prefix="/api/payments", tags=["支付"])


@router.post("/create-order", response_model=CreateOrderOut)
async def create_order(
    member_id: str,
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    result = await db.execute(
        select(Member).where(Member.id == member_id, Member.user_id == current_user.id)
    )
    member = result.scalar_one_or_none()
    if member is None:
        raise HTTPException(status_code=404, detail="会员不存在")

    existing = await db.execute(
        select(Order).where(Order.member_id == member_id, Order.status == "paid")
    )
    if existing.scalar_one_or_none():
        raise HTTPException(status_code=400, detail="该会员已付费")

    order_no = generate_order_no()
    order = Order(
        user_id=current_user.id,
        member_id=member_id,
        order_no=order_no,
        total_fee=settings.PLAN_PRICE,
    )
    db.add(order)
    await db.commit()
    await db.refresh(order)

    prepay_id = f"mock_{order_no}"
    order.prepay_id = prepay_id
    await db.commit()

    pay_params = get_miniapp_pay_params(prepay_id)

    return CreateOrderOut(
        order_id=order.id,
        order_no=order_no,
        total_fee=order.total_fee,
        prepay_id=prepay_id,
        pay_params=pay_params,
    )


@router.post("/notify")
async def pay_notify(notify: PayNotify):
    return {"return_code": "SUCCESS", "return_msg": "OK"}


@router.get("/orders", response_model=list[OrderOut])
async def list_orders(
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    result = await db.execute(
        select(Order).where(Order.user_id == current_user.id).order_by(Order.created_at.desc())
    )
    orders = result.scalars().all()
    return [
        OrderOut(
            id=o.id, order_no=o.order_no, total_fee=o.total_fee,
            status=o.status, paid_at=o.paid_at, created_at=o.created_at,
        )
        for o in orders
    ]
